framework_tvb/tvb/interfaces/web/controllers/flow_controller.py

```diff
@@ -241,69 +245,43 @@ def default(self, step_key, adapter_key, cancel=False, back_page=None, not_reset
             template_specification[common.KEY_DISPLAY_MENU] = True
             template_specification[common.KEY_BACK_PAGE] = back_page_link
 
+        common.add2session(common.KEY_ADAPTER, adapter_key)
+
         template_specification[common.KEY_ADAPTER] = adapter_key
         template_specification[ABCDisplayer.KEY_IS_ADAPTER] = True
         self.fill_default_attributes(template_specification, algorithm.displayname)
         return template_specification
 
-    # @expose_fragment("flow/type2component/datatype2select_simple")
-    def getfiltereddatatypes(self, name, parent_div, tree_session_key, filters):
-        # TODO: fix this use-case
+    @expose_fragment('form_fields/options_field')
+    @settings
+    @context_selected
+    def get_filtered_datatypes(self, datatype_index_path, filters):
```

Comment:
Wouldn't be simpler the code if you would pass the `dt_module` and `dt_class` strings here?

Reply:
Done.